tmp/simple_solver.py

```diff
@@ -0,0 +1,54 @@
+######################
+# Python Simple Solver
+######################
+
+from itertools import combinations
+from typing import NamedTuple, List, Dict, Set
+from event_listener.dfusion_db.models import Order
+
+
+class TradeExecution(NamedTuple):
+    buy_amount: int = 0
+    sell_amount: int = 0
+
+
+class Solution(NamedTuple):
+    prices: Dict[int, int]
+    amounts: Dict[Order, TradeExecution]
+
+
+def simple_solve(orders: List[Order], tokens: Set[int]) -> Solution:
+    res_price = {t: 1 for t in tokens}
+    res_vol = {o: TradeExecution() for o in orders}
+
+    for x, y in list(combinations(orders, r=2)):
+
+        match_conditions = [
+            x.buy_token == y.sell_token,
+            y.buy_token == x.sell_token,
+            x.buy_amount * y.buy_amount <= y.sell_amount * x.sell_amount
+        ]
+
+        if all(match_conditions):
+
+            if x.buy_amount <= y.sell_amount and x.sell_amount <= y.buy_amount:  # Type I-A (x <= y)
+                res_price[x.buy_token] = x.sell_amount
```

Comment:
What prevents an order from getting executed multiple times? E.g. order 2 could be matched first with order 1 and then order 2?